fetch.bs

```diff
@@ -2128,29 +2128,31 @@ given <var>request</var>, run these steps:
  <li><p>Let <var>topLevelOrigin</var> be null.
 
  <li>
-  <p>If <var>request</var>'s <a for=request>client</a> is non-null, then:
+  <p>If <var>request</var>'s <a for=request>reserved client</a> is non-null, then:
 
   <ol>
-   <li><p>Set <var>topLevelOrigin</var> to <var>request</var>'s <a for=request>client</a>'s
+   <li><p>Set <var>topLevelOrigin</var> to <var>request</var>'s <a for=request>reserved client</a>'s
    <a for="environment">top-level origin</a>.
 
+   <li>
+    <p>If <var>topLevelOrigin</var> is null, then set <var>topLevelOrigin</var> to
+    <var>request</var>'s <a for=request>reserved client</a>'s
+    <a for="environment">top-level creation URL</a>'s <a for=url>origin</a>.
+
+    <p class=note>This happens for top-level navigations only.
+
    <li><p>Assert: <var>topLevelOrigin</var> is an <a for=/>origin</a>.
```

Comment:
Since topLevelOrigin can be null here, is it ok to have this assert for null?

Reply:
Isn't that being checked in 2148

Reply:
That only applies if the reserved client was null, but I think I can maybe improve this a bit by using otherwise more.